In [1]:
import numpy as np
array1 = np.array([2, 3, 5, 7, 11, 13, 17, 19, 31])

In [2]:
array1>4

array([False, False,  True,  True,  True,  True,  True,  True,  True])

In [3]:
array1 % 2 == 0

array([ True, False, False, False, False, False, False, False, False])

### Boolean 값이 담긴 numpy array 를 만들어보았다
### 이번에는 numbpyarray를 만들어보자.

In [9]:
booleans = np.array([True, True, False, False, True])

In [12]:
np.where(booleans)

(array([0, 1, 4], dtype=int64),)

## 문제
## 영훈이가 창업한 흥부부대찌개 신주쿠점은 이제 직장인들에게 소문난 맛집입니다. 그러나 매일같이 성공적인 것은 아닙니다. 목표 일 매출을 달성하지 못하는 날들이 아직 꽤 있거든요. 영훈이가 생각하는 성공적인 하루 매출은 20만 엔입니다.

## 성공적이지 않은 날의 매출만 골라서 보고 싶습니다. 20만 엔 이하의 매출만 담긴 numpy array를 출력해주세요.
## 반복문은 사용하면 안 됩니다!

## 주의 사항: 자동 채점 과제입니다. 정답 출력 코드는 print 없이 작성해 주세요. (예시: bad_days_revenue)

In [30]:
import numpy as np

revenue_in_yen = [
    300000, 340000, 320000, 360000, 
    440000, 140000, 180000, 340000, 
    330000, 290000, 280000, 380000, 
    170000, 140000, 230000, 390000, 
    400000, 350000, 380000, 150000, 
    110000, 240000, 380000, 380000, 
    340000, 420000, 150000, 130000, 
    360000, 320000, 250000
]

# 풀이1
Today_I_Earned = np.array(revenue_in_yen)
filter = np.where(Today_I_Earned<200000)
filter #여기까지하고 filter확인하면 (array([ 5,  6, 12, 13, 19, 20, 26, 27]),) 이게나옴
Today_I_Earned = Today_I_Earned[filter]
Today_I_Earned 

# 풀이2

yen_array = np.array(revenue_in_yen)
bad_days_revenue = yen_array[yen_array<= 200000]
bad_days_revenue



array([140000, 180000, 170000, 140000, 150000, 110000, 150000, 130000])

---
### 최대값

In [31]:
import numpy as np

array1 = np.array([14, 6, 13, 21, 23, 31, 9, 5])

print(array1.max()) #최대값 구하기
print(array1.min()) #최소값 구하기


31
5


### 최소값

In [33]:
import numpy as np

array1 = np.array([14, 6, 13, 21, 23, 31, 9, 5])

print(array1.mean()) #평균값 구하기

15.25


## 중앙값
#### median 메소드를 사용하면 중간값을 구할 수 있는데요. 특이하게 median은 numpy array의 메소드가 아니라 numpy의 메소드입니다.

In [38]:
import numpy as np

array1 = np.array([8, 12, 9, 15, 16])
array2 = np.array([14, 6, 13, 21, 23, 31, 9, 5])

print(np.median(array1)) #중앙값
print(np.median(array2)) #중앙값

12.0
13.5


#### array1을 정렬하면 중앙값이 12 입니다.
#### array2에는 짝수개의 요소가 있기 때문에 중앙값이 13과 14 두개 둘의 평균은 13.5

---
## 표준 편차, 분산

In [41]:
import numpy as np

array1 = np.array([14, 6, 13, 21, 23, 31, 9 ,5])

print(array1.std()) #표준 편차
print(array1.var()) #분산

8.496322733983215
72.1875
